pretty_midi/instrument.py

```diff
@@ -430,57 +431,77 @@ def synthesize(self, fs=44100, wave=np.sin):
         """Synthesize using fluidsynth.
 
         Parameters
         ----------
         fs : int
-            Sampling rate to synthesize.
+            Sampling rate to synthesize at.
+            Only used when a new instance of fluidsynth.Synth is created.
+            Default ``44100``.
+        synthesizer : fluidsynth.Synth or str
+            fluidsynth.Synth instance to use or a string with the path to a .sf2 file.
+            Default ``None``, which creates a new instance using the TimGM6mb.sf2 file
+            included with ``pretty_midi``.
+        sfid : int
+            Soundfont ID to use if an instance of fluidsynth.Synth is provided.
+            Default ``0``, which uses the first soundfont.
         sf2_path : str
             Path to a .sf2 file.
             Default ``None``, which uses the TimGM6mb.sf2 file included with
             ``pretty_midi``.
+            .. deprecated:: 0.2.11
+                Use :param:`synthesizer` instead.
 
         Returns
         -------
         synthesized : np.ndarray
             Waveform of the MIDI data, synthesized at ``fs``.
 
         """
-        # If sf2_path is None, use the included TimGM6mb.sf2 path
-        if sf2_path is None:
-            sf2_path = pkg_resources.resource_filename(__name__, DEFAULT_SF2)
-
         if not _HAS_FLUIDSYNTH:
-            raise ImportError("fluidsynth() was called but pyfluidsynth "
-                              "is not installed.")
+            raise ImportError("fluidsynth() was called but pyfluidsynth is not installed.")
+
+        if sf2_path is not None:
+            warn("The parameter 'sf2_path' is deprecated, please use 'synthesizer' instead.",
+                 DeprecationWarning, 2)
+            if synthesizer is not None:
+                raise ValueError("sf2_path and synthesizer cannot both be supplied.")
+            else:
+                synthesizer = sf2_path
 
-        if not os.path.exists(sf2_path):
-            raise ValueError("No soundfont file found at the supplied path "
-                             "{}".format(sf2_path))
+        if synthesizer is None:
+            synthesizer = pkg_resources.resource_filename(__name__, DEFAULT_SF2)
 
         # If the instrument has no notes, return an empty array
         if len(self.notes) == 0:
             return np.array([])
 
-        # Create fluidsynth instance
-        fl = fluidsynth.Synth(samplerate=fs)
-        # Load in the soundfont
-        sfid = fl.sfload(sf2_path)
+        # Create a fluidsynth instance if one wasn't provided
+        if type(synthesizer) is str:
+            sf2_path = synthesizer
+            if not os.path.exists(synthesizer):
+                raise ValueError("No soundfont file found at the supplied path {}".format(sf2_path))
+            synthesizer = fluidsynth.Synth(samplerate=fs)
+            delete_synthesizer = True
+            sfid = synthesizer.sfload(sf2_path)
+        else:
```

Comment:
I'd suggest making this 
`elif isinstance(synthesizer, fluidsynth.Synth):`
and then later doing
```
else:
    raise ValueError("synthesizer must be a str or fluidsynth.Synth instance"
````